arm9/src/unk_0205FA2C.c

```diff
@@ -0,0 +1,546 @@
+#include "unk_0205FA2C.h"
+
+extern void *UNK_020F96DC;
+extern void *UNK_020FA6E8;
+extern u32 FUN_02079C70(struct SaveBlock2 *sav2);
+extern void FUN_0207B000(struct UnkPlayerStruct2_0205FA2C *ptr, const u8 param1[12]);
+extern void FUN_0207C2A4(struct UnkPlayerStruct2_0205FA2C *ptr, struct PlayerData *player_data);
+extern u32 FUN_0203384C(u32 *param0);
+extern u32 *FUN_02038790(struct UnkStruct_0204639C *param0, u16 param1, u16 param2);
+extern u16 *GetVarPointer(struct UnkSavStruct80 *arg, u16);
+extern u32 FUN_02031190();
+extern u32 FUN_020316E0(u32 param0);
+extern u16 MOD06_02244660(struct UnkStruct_0204639C *param0, u32 param1);
+extern u16 MOD06_022446BC(struct UnkStruct_0204639C *param0, u32 param1);
+extern u16 MOD06_022446E0(struct UnkStruct_0204639C *param0, u32 param1);
+extern void FUN_0202A5CC(u32 param0, u32 param1);
+extern u32 FUN_0202A5D0(u32 param0);
+extern u32 FUN_0202A150(struct UnkStruct_02029FB0 *param0, u32 param1);
+extern u32 FUN_0202A8D8(struct SaveBlock2 *sav2);
+extern u32 FUN_0202A578(u32 param0, u32 param1, u32 param2);
+extern u32 FUN_02026CC4(struct SaveBlock2 *sav2);
+extern u32 FUN_02025D94(u32 param0, u32 param1);
+extern u32 FUN_0202A8CC(struct SaveBlock2 *sav2);
+extern void FUN_0202A2C4(u32 param0, u32 param1, u32 *param2);
+extern u32 FUN_0202A240(u32 param0, u32 param1, u32 param2);
+extern int FUN_0202A538(u32 param0, u16 param1, u32 param2);
+
+const u8 UNK_020F7454[] = {
+    0x00,
+    0x01,
+    0x02,
+    0x04,
+    0x03,
+    0x05,
+    0x06,
+    0x07,
+    0x08,
+    0x00,
```

Comment:
This and the following two are implicit padding values for file boundary word alignment.

Reply:
Ah interesting. Is this always the case for constants at the top of a file?

Reply:
It's the case when the object immediately following the array has some alignment forced upon it which is not satisfied by simply placing the new object at the end of the previous one.  For example, when a `u8[9]` is followed by a `u32`. The u32 needs to be 4-bytes aligned on ARM, so 0s are placed to fill the gap between the u8 array and the u32.

Reply:
It makes sense, thanks. Here the array is followed by a function. I guess there are 4-bytes aligned too?

Reply:
Not exactly. The order they appear in the C file is not the same as the order they appear in the final binary. Data arrays are placed in .rodata sections whereas functions are placed in .text sections.